### PR TITLE
Add support for one-time URL flags

### PR DESCRIPTION
**Pull request by dmnks**

### CODE REVIEW

**Comment by dmnks** [comment]
Hi Jarda, could you please review this one? (depending on your availability of course)

**Comment by dmnks** [comment]
libdnf change that makes use of this new feature:
https://github.com/rpm-software-management/libdnf/pull/807

**Comment by Conan-Kudo** [comment]
@dmnks Can you please rebase?

**Comment by dmnks** [comment]
Yup, will do, but I have a suspicion that the code change causing this conflict was actually unintentional, so I'll check with the team tomorrow and then rebase accordingly.

**Comment by dmnks** [comment]
FTR, after talking to the team, the librepo change turns out to be correct (it's a temporary workaround) so I'll proceed with the rebase.

**Comment by dmnks** [comment]
Rebased.

File: librepo/downloader.c [hunk 1]
```diff
@@ -1464,6 +1464,20 @@ prepare_next_transfer(LrDownload *dd, gboolean *candidatefound, GError **err)
     }
     target->curl_handle = h;
 
+    // Append the LRO_ONETIMEFLAG if instructed to do so
+    LrHandle *handle = target->handle;
+    if (handle && handle->onetimeflag && handle->otf_context) {
+        char *sep = "?";
+        if (g_strrstr(full_url, sep) != NULL) sep = "&";
```
Comment (Conan-Kudo) [inline librepo/downloader.c]: Can we please not do oneliner `if` conditionals, they're hard to read...

Comment (dmnks) [reply librepo/downloader.c]: Fixed.

**Pull request merged**